src/modules/BuilderPlace/actions.ts

```diff
@@ -432,47 +844,63 @@ export async function checkOrResetTransactionCounter(
 }
 
 export async function incrementWeeklyTransactionCounter(
-  worker: IWorkerMongooseSchema,
+  user: User,
   res: NextApiResponse,
 ): Promise<void> {
   try {
-    await connection();
     // Increment the counter
-    worker.weeklyTransactionCounter = (worker.weeklyTransactionCounter || 0) + 1;
-    console.log('Transaction counter incremented', worker.weeklyTransactionCounter);
-    await worker.save();
+    const newWeeklyTransactionCounter = (user.weeklyTransactionCounter || 0) + 1;
+    await prisma.user.update({
+      where: {
+        id: user.id,
+      },
+      data: {
+        weeklyTransactionCounter: newWeeklyTransactionCounter,
+      },
+    });
+    console.log('Transaction counter incremented', newWeeklyTransactionCounter);
   } catch (error: any) {
     res.status(500).json({ error: error.message });
   }
 }
 
-export const verifyWorkerProfileEmail = async (id: string) => {
+export const verifyUserEmail = async (id: string, res?: NextApiResponse) => {
+  let errorMessage;
   try {
-    await connection();
-    const existingWorker = await Worker.findOne({ _id: id });
-    if (existingWorker) {
-      const resp = await Worker.updateOne({ _id: id }, { emailVerified: true }).exec();
-      if (resp.modifiedCount === 0 && resp.matchedCount === 1) {
-        console.log(EMAIL_ALREADY_VERIFIED);
-        return {
-          error: EMAIL_ALREADY_VERIFIED,
-        };
-      }
-      console.log('Updated worker profile email', resp);
-    } else {
-      console.log(ERROR_VERIFYING_EMAIL);
-      return {
-        error: ERROR_VERIFYING_EMAIL,
-      };
+    const existingUser = await prisma.user.findUnique({
+      where: {
+        id: Number(id),
+      },
+    });
+    if (existingUser?.isEmailVerified === true) {
+      console.log(EMAIL_ALREADY_VERIFIED);
+      throw new Error(EMAIL_ALREADY_VERIFIED);
     }
+    const resp = await prisma.user.update({
+      where: {
+        id: Number(id),
+      },
+      data: {
+        isEmailVerified: true,
+      },
+    });
+    console.log('Validated user email', resp.id, resp.name, resp.email);
     return {
       message: EMAIL_VERIFIED_SUCCESSFULLY,
-      email: existingWorker.email,
+      email: resp.email,
     };
   } catch (error: any) {
-    console.log(error.message);
-    return {
-      error: error.message,
-    };
+    if (error?.name?.includes('Prisma')) {
+      errorMessage = ERROR_VERIFYING_EMAIL;
+    } else {
+      errorMessage = error.message;
+    }
+    if (res) {
+      console.log(error.message);
+      res.status(500).json({ error: errorMessage });
+    } else {
+      console.log(error.message);
+      throw new Error(ERROR_VERIFYING_EMAIL);
+    }
```

Comment:
To do at the end when everything work fine: split this file into one file by entity